// material-ui/Hidden/Hidden.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _HiddenJs = require('./HiddenJs');

var _HiddenJs2 = _interopRequireDefault(_HiddenJs);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var babelPluginFlowReactPropTypes_proptype_HiddenProps = require('./types').babelPluginFlowReactPropTypes_proptype_HiddenProps || require('prop-types').any;

/**
 * Responsively hides children based on the selected implementation.
 */
function Hidden(props) {
  var componentProp = props.component,
      implementation = props.implementation,
      other = (0, _objectWithoutProperties3.default)(props, ['component', 'implementation']);

  // workaround: see https://github.com/facebook/flow/issues/1660#issuecomment-297775427

  var component = componentProp || Hidden.defaultProps.component;

  if (implementation === 'js') {
    return _react2.default.createElement(_HiddenJs2.default, (0, _extends3.default)({}, other, { component: component }));
  }

  throw new Error('<Hidden implementation="css" /> is not yet implemented');
}

Hidden.defaultProps = {
  component: 'div',
  implementation: 'js',
  xsUp: false,
  smUp: false,
  mdUp: false,
  lgUp: false,
  xlUp: false,
  xsDown: false,
  smDown: false,
  mdDown: false,
  lgDown: false,
  xlDown: false
};

exports.default = Hidden;
